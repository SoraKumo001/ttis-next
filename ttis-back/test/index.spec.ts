import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import { AppModule } from '../src/app.module';
import fetch from 'isomorphic-unfetch';
import * as fs from 'fs';
import * as path from 'path';

import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from 'apollo-boost';
import { AuthLink, createAuthLink, beforeSync } from './testTools';

let app: INestApplication;
let port: number;
export class CustomApolloClient extends ApolloClient<NormalizedCacheObject> {
  link: AuthLink;
  constructor() {
    const link: AuthLink = createAuthLink({
      fetch,
      uri: `http://localhost:${port}/graphql`,
    });
    super({ link, cache: new InMemoryCache().restore({}) });
  }
  setToken(token: string) {
    this.link.setToken(token);
  }
}

const execTest = (dir: string = __dirname) => {
  const files = fs.readdirSync(dir);
  files.forEach((file) => {
    const filePath = path.resolve(dir, file);
    const state = fs.statSync(filePath);
    if (state.isDirectory()) execTest(filePath);
    else /.*\.test\.ts$/.test(file) && require(filePath);
  });
};

export const before = beforeSync(async () => {
  const module: TestingModule = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();
  app = module.createNestApplication();
  await app.init();
  await app.listen(0);
  port = app.getHttpServer().address().port;
});
export const createClient = async()=>{
  await before;
  return new CustomApolloClient();
}
afterAll(async () => {
  await app.close();
});

describe('Test', () => {
  execTest();
});
