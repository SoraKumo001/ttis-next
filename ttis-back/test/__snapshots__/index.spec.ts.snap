// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test User(login) CreateUser 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "__typename": "User",
      "id": 2,
      "info": "{}",
      "name": "Test01",
    },
  },
}
`;

exports[`Test User(login) CurrentUser 1`] = `
Object {
  "data": Object {
    "currentUser": Object {
      "__typename": "User",
      "id": 1,
      "info": "{}",
      "name": "admin",
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Test User(login) UpdateUser 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "__typename": "User",
      "id": 2,
      "info": "{}",
      "name": "Test01-02",
    },
  },
}
`;

exports[`Test User(login) User List 1`] = `
Array [
  Object {
    "__typename": "User",
    "id": 1,
    "info": "{}",
    "name": "admin",
  },
]
`;
